### DISCUSSION TITLE
CORECTL Support

### DISCUSSION THREAD

**Discussion by thetechhimself** [osy86/HaC-Mini#1017] [General]
It appears our native Linux counterparts have an AMD acceleration control open-source option. It can conceivably be built on MacOS with Hombrew-CMake however can the necessary boot flag (amdgpu.ppfeaturemask=0xffffffff) be set in the bootloader? Also, since this hasn't been tested before (that I'm aware of), I'm unaware if there are any other dependency conflicts e.g. Desktop Environment, existing Apple Driver implementation flag locks.

#Setup for CORECTRL; boot loader injection example
https://gitlab.com/corectrl/corectrl/-/wikis/Setup

#CORECTRL URL
https://gitlab.com/corectrl/corectrl

**Comment by osy** [comment]
So unless I’m missing something here, you want to run an app written for Linux on macOS? This is like trying to run a PlayStation game on Xbox. There’s a lot of work involved from userland to kernel to drivers.

**Reply by thetechhimself** [reply]
You didn't. I'm unfamiliar with the kernel and drivers differences between Linux and macOS for AMD graphics.

Not suggesting you do it either.

Was curious if our Linux counterparts were using a flavor of the same driver as macOS that could be repurposed by injecting the flag into the boot loader and compiling it natively. Figured I'd ask as if it was that easy, this probably would've already been done.

While I have you, my thanks for doing this project. Hades Canyon is a wonderful platform that can run an array of older macOS's (with your and other's help) which have their unique benefits and is a cost effective solution for a low profile, high performance desktop that's still efficient by modern standards. Hades Canyon is truly a mesh of the best of technologies at the time, being able to run the best OS possible on it, sweetens the deal. To say nothing of the support other Linux platforms have for the platform.
